spec/models/league_spec.rb

```diff
@@ -140,6 +141,146 @@
         expect(subject.length).to eq(league.teams.length * (league.teams.length - 1))
       end
     end
+
+    context 'swiss system' do
+      let(:gamemode) { League.game_modes[:swiss] }
+
+      it 'does not have an up-to-date schedule if the last gameday is in the past' do
+        league.gamedays << FactoryBot.create(:gameday)
+
+        last_gameday = league.gamedays.last
+        last_gameday.starttime = 2.days.ago
+        last_gameday.endtime = 1.day.ago
+
+        expect(league.is_up_to_date).to be false
+      end
+
+      it 'has an up-to-date schedule if the last gameday is not over yet' do
+        league.gamedays << FactoryBot.create(:gameday)
+
+        last_gameday = league.gamedays.last
+        last_gameday.starttime = Date.today.next_day
+        last_gameday.endtime = Date.today.next_day 2
+
+        expect(league.is_up_to_date).to be true
+      end
+
+      it 'allows to check if two teams have already played a match' do
+        teams = FactoryBot.create_list(:team, 2)
+
+        expect(league.have_already_played(teams[0], teams[1])).to be false
+        expect(league.have_already_played(teams[1], teams[0])).to be false
+
+        league.matches << Match.new(team_home: teams[0], team_away: teams[1], gameday_number: 0)
+
+        expect(league.have_already_played(teams[0], teams[1])).to be true
+        expect(league.have_already_played(teams[1], teams[0])).to be true
+      end
+
+      context 'update schedule' do
+        it 'creates a new gameday' do
+          gameday_amount = league.gamedays.length
+          league.update_schedule
+          expect(league.gamedays.length).to be gameday_amount + 1
+        end
+
+
+        it 'creates a correct amount of matches' do
+          matches_amount = league.matches.length
+          league.update_schedule
+          expect(league.matches.length).to be matches_amount + (league.teams.length / 2).floor
```

Comment:
This test only checks for one gameday with one constellation of Teams. The problem I mentioned earlier isn't tested this way.